spec/helpers/application_helper_spec.rb

```diff
@@ -29,15 +29,21 @@
 
   describe 'body_classes' do
     context 'with a body class string from a controller' do
-      before do
-        without_partial_double_verification do
-          allow(helper).to receive_messages(body_class_string: 'modal-layout compose-standalone', current_theme: 'default', current_account: Fabricate(:account))
-        end
-      end
+      before { helper.extend controller_helpers }
 
       it 'uses the controller body classes in the result' do
         expect(helper.body_classes).to match(/modal-layout compose-standalone/)
       end
+
+      private
+
+      def controller_helpers
+        Module.new do
+          def body_class_string = 'modal-layout compose-standalone'
+          def current_account = Fabricate(:account)
```

Comment:
Wouldn't this `Fabricate` a new `Account` on each call to `current_account`? This could cause issues down the road.

Reply:
Practically speaking I think it only gets called once during the spec - but I see your point, updated to memoize.